File: ui/interfaces/investigation.ts
// src/types/investigation.ts
export interface Investigation {
    id: string; // UUID as string
    incident_id: string; // UUID as string
    lead_investigator_id: string; // UUID as string
    root_cause: string;
    contributing_factors: Record<string, any>; // JSONB field
    investigation_methods: Record<string, any>; // JSONB field
    findings: string;
    recommendations: string;
    started_at: string; // ISO date string
    completed_at?: string; // Optional ISO date string
    status: 'in_progress' | 'pending_review' | 'completed' | 'reopened';
    created_at: string; // ISO date string
    updated_at: string; // ISO date string
  }

File: internal/models/Investigation.go
package models

import (
	"time"

	"github.com/google/uuid"
)

type Investigation struct {
	ID                   uuid.UUID `gorm:"type:uuid;primaryKey;default:uuid_generate_v4()"`
	IncidentID           uuid.UUID `gorm:"type:uuid;not null;unique"`
	LeadInvestigatorID   uuid.UUID `gorm:"type:uuid;not null"`
	Description          string    `gorm:"type:text"`
	RootCause            string    `gorm:"type:text"`
	ContributingFactors  JSONB     `gorm:"type:jsonb"`
	InvestigationMethods JSONB     `gorm:"type:jsonb"`
	Findings             string    `gorm:"type:text"`
	Recommendations      string    `gorm:"type:text"`
	StartedAt            time.Time `gorm:"not null"`
	CompletedAt          *time.Time
	Status               string    `gorm:"size:30;not null;default:'in_progress';check:status IN ('in_progress', 'pending_review', 'completed', 'reopened')"`
	CreatedAt            time.Time `gorm:"default:CURRENT_TIMESTAMP"`
	UpdatedAt            time.Time `gorm:"default:CURRENT_TIMESTAMP"`

	// Relationships
	Incident         Incident `gorm:"foreignKey:IncidentID"`
	LeadInvestigator Employee `gorm:"foreignKey:LeadInvestigatorID"`
}
